source/common/formatter/substitution_format_string.cc

```diff
@@ -25,5 +26,13 @@ FormatterPtr SubstitutionFormatStringUtils::fromProtoConfig(
   return nullptr;
 }
 
+absl::string_view SubstitutionFormatStringUtils::getContentType(
+    const envoy::config::core::v3::SubstitutionFormatString& config) {
+  if (config.content_type() == Http::Headers::get().ContentTypeValues.Html) {
+    return Http::Headers::get().ContentTypeValues.Html;
+  }
```

Comment:
Not sure, but can we pass `config.content_type()` value as is? And yes, when it is not set and it is not JSON we return `text/plain`. WDYT @htuch?

Reply:
@dio - I tried doing that initially, however `config.content_type()` return `std:string&` and after the value would get assigned to `absl:string_view` content_type_ , it would loose that reference object. My understanding about string_view is that it's a read-only pointer to the string object. One way to fix that was to change the datatype of `content_type_` to `std::string`, however I wasn't sure if that is encouraged ? Any other ideas ?

Reply:
Sure, make `content_type_` a `std::string`.